src/v/cluster/partition.cc

```diff
@@ -426,6 +429,21 @@ partition::timequery(storage::timequery_config cfg) {
     cfg.max_offset = _raft->get_offset_translator_state()->to_log_offset(
       cfg.max_offset);
     result = co_await _raft->timequery(cfg);
+
```

Comment:
this looks strange to me, the alg. here is to
1. read from the cloud storage 
2. fall through if no data is returned and read from raft
3. without checking the result read from the cloud storage once again and possibly return the result
4. take result of 2 into account

maybe line 432 need to be moved below the next `if` branch?